src/tensor.cpp

```diff
@@ -331,15 +331,13 @@ const std::vector<std::size_t>& TensorDescriptor::GetLengths() const { return le
 
 const std::vector<std::size_t>& TensorDescriptor::GetStrides() const { return strides; }
 
-int TensorDescriptor::GetSize() const
-{
-    assert(lens.size() == strides.size());
-    return lens.size();
-}
+unsigned TensorDescriptor::GetNumDims() const { return lens.size(); }
+
+// This name is misleading, all the usages should be replaced with GetNumDims()
+int TensorDescriptor::GetSize() const { return GetNumDims(); }
 
 std::size_t TensorDescriptor::GetElementSize() const
 {
-    assert(lens.size() == strides.size());
```

Comment:
What was wrong with this assert?

Reply:
It is already checked in the constructor.